streampipes-client-python/streampipes/functions/broker/output_collector.py

```diff
@@ -32,14 +32,14 @@ class OutputCollector:
 
     Attributes
     ----------
-    broker: Broker
-        The broker instance that sends the data to StreamPipes
+    publisher: Publisher
+        The publisher instance that sends the data to StreamPipes
 
     """
 
     def __init__(self, data_stream: DataStream) -> None:
-        self.broker = get_broker(data_stream)
-        self._run_coroutine(self.broker.connect(data_stream))
+        self.publisher: Publisher = get_broker(data_stream, is_publisher=True)  # type: ignore
```

Comment:
Can't you get rid of `# type: ignore` here?

Reply:
Maybe, but `get_broker` returns a `Publisher` or `Subscriber`. Do you know a solution for this?

Reply:
We could use a `cast` like this:
```
self.publisher = cast(Publisher, get_broker(data_stream, is_publisher=True))
```
Do you think this is a better solution or does it make the code less readable?

Reply:
If it's only about two return types, you can do `get_broker(...) -> Union[Publisher | Subscriber]` 
But they have the same parent class right?
So in this case the parent class should work as return type as well

Reply:
Yes, the return type is the parent class `Broker`, but mypy shows the error:
```
Incompatible types in assignment (expression has type "Broker", variable has type "Publisher")
```

Reply:
Okay, then let's just stick with `type:ignore`